kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/actions/WorkLockAction.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+package org.kitodo.mediaserver.core.actions;
+
+import java.util.Map;
+import org.kitodo.mediaserver.core.api.IAction;
+import org.kitodo.mediaserver.core.db.entities.Work;
+import org.kitodo.mediaserver.core.db.repositories.WorkRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Locks or unlocks a work.
+ */
+public class WorkLockAction implements IAction {
+
+    private WorkRepository workRepository;
+
+    @Autowired
+    public void setWorkRepository(WorkRepository workRepository) {
+        this.workRepository = workRepository;
+    }
+
+    /**
+     * Locks or unlocks the work.
+     * @param work      a work entity
+     * @param parameter a map of parameter; enabled: lock or unlock, comment: lock comment
+     * @return always null
+     * @throws Exception no exception thrown in this implementation
+     */
+    @Override
+    public Object perform(Work work, Map<String, String> parameter) throws Exception {
+        Boolean enabled = Boolean.parseBoolean(parameter.get("enabled"));
+        work.setEnabled(enabled);
```

Comment:
This will break with an NPE if the parameter map is null. Also it will always disable the work if the parameter "enabled" is not set in the map, with no questions asked. Suggestion: use the method checkForRequiredParameter in MedaServerUtils to check that the param is set and the map not null (just added the null check, please fetch...)